### PR TITLE
APEXCORE-719 Pass an application name from stram client to applicatio…

### PR DESCRIPTION
**Pull request by sgolovko**
…n master and container via command line properties

Added passing of the application name via command line properties

@vrozov Could you please review and merge the PR?

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java [hunk 1]
```diff
@@ -253,6 +253,7 @@ public void run()
     vargs.add(String.format("-D%scid=%s", StreamingApplication.DT_PREFIX, jvmID));
     vargs.add("-Dhadoop.root.logger=" + (dag.isDebug() ? "DEBUG" : "INFO") + ",RFA");
     vargs.add("-Dhadoop.log.dir=" + ApplicationConstants.LOG_DIR_EXPANSION_VAR);
+    vargs.add(String.format("-D%s='%s'", LogicalPlan.APPLICATION_NAME.getLongName(), dag.getValue(LogicalPlan.APPLICATION_NAME)));
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java]: Application name may contain `'` and possibly other characters that need to be escaped

Comment (sgolovko) [reply engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java]: Fixed.

Per man bash:

Enclosing characters in single quotes preserves the literal value of each character within the quotes. A single quote may not occur between single quotes, even when preceded by a backslash.

Comment (sgolovko) [reply engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java]: Added checking of special symbols (', ", $, \\) in application names.

File: engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java [hunk 2]
```diff
@@ -334,4 +336,23 @@ public static ByteBuffer getTokens(UserGroupInformation ugi, Token<StramDelegati
     }
   }
 
+  public static String addBackSlashBeforeSpecialSymbols(String string)
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java]: What is the reason not to use a regular expression? For example:
```
string.replaceAll("['\"$\\\\]", "\\\\$0");
```

File: engine/src/main/java/com/datatorrent/stram/StramClient.java [hunk 3]
```diff
@@ -572,6 +572,8 @@ public void startApplication() throws YarnException, IOException
       vargs.add("-Dhadoop.root.logger=" + (dag.isDebug() ? "DEBUG" : "INFO") + ",RFA");
       vargs.add("-Dhadoop.log.dir=" + ApplicationConstants.LOG_DIR_EXPANSION_VAR);
       vargs.add(String.format("-D%s=%s", StreamingContainer.PROP_APP_PATH, dag.assertAppPath()));
+      vargs.add(String.format("-D%s=$'%s'", LogicalPlan.APPLICATION_NAME.getLongName(),
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/StramClient.java]: To avoid code duplication, introduce static helper method in StramClientUtils.java.

Comment (sgolovko) [reply engine/src/main/java/com/datatorrent/stram/StramClient.java]: Done

File: engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java (hunk 2 above)
Comment (sgolovko) [reply engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java]: Done.

File: engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java [hunk 4]
```diff
@@ -870,4 +870,8 @@ public static String getSocketConnectString(InetSocketAddress socketAddress)
     return appInfo;
   }
 
+  public static String addBackSlashBeforeSpecialSymbols(String string)
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]:
```
  public static void addAttributeToArgs(Attribute<String> attribute, Context context, List<CharSequence> vargs)
  {
    if (context.getValue(attribute) != null) {
      vargs.add(String.format("-D%s=$'%s'", attribute.getLongName(), context.getValue(attribute).replaceAll("['\"$\\\\]", "\\\\$0")));
    }
  }
```

Comment (sgolovko) [reply engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java]: Agree. Done.

**Pull request merged**